crates/bevy_ecs/src/event.rs

```diff
@@ -159,7 +159,7 @@ fn map_instance_event<T>(event_instance: &EventInstance<T>) -> &T {
 
 /// Reads events of type `T` in order and tracks which events have already been read.
 #[derive(SystemParam)]
-pub struct EventReader<'w, 's, T: Resource> {
+pub struct EventReader<'w, 's, T: Resource + Sync> {
```

Comment:
I dislike these bounds here. 

The requested resource is not actually `T`, it's `Events<T>`.  The problem is that this works somewhat incidentally, because `Events<T>: Resource + Sync` if and only if `T: Resource + Sync`. 

I think I'll open up a PR to finally make an `Event` trait.

Reply:
Said PR: #4685. I'm a bit reluctant to merge this in before that; I don't like the way that this is randomly mucking with the `Event` types.

Reply:
Updated this PR to include those changes. Events no longer are affected by this change.